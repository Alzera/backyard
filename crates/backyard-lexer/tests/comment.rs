use backyard_lexer::lex;

#[test]
fn line() {
  let tokens = lex(true, "// test").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn line_new_line() {
  let tokens = lex(true, "// test\n$a").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn line_tag_close() {
  let tokens = lex(false, "<?php // test ?>").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn block() {
  let tokens = lex(true, "/* test */").unwrap();
  for token in tokens.iter() {
    println!("{token:?}");
  }
  println!("");
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_1() {
  let tokens = lex(true, "/** @param string $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_2() {
  let tokens = lex(true, "/** @param   string   $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_3() {
  let tokens = lex(true, "/** @param  ( string )  $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_4() {
  let tokens = lex(true, "/** @param ( ( string ) ) $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_5() {
  let tokens = lex(true, "/** @param \\\\Foo\\Bar\\\\Baz $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_6() {
  let tokens = lex(true, "/** @param   \\\\Foo\\Bar\\\\Baz   $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_7() {
  let tokens = lex(true, "/** @param  ( \\\\Foo\\Bar\\\\Baz )  $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_8() {
  let tokens = lex(true, "/** @param ( ( \\\\Foo\\Bar\\\\Baz ) ) $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_9() {
  let tokens = lex(true, "/** @param string|int $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_10() {
  let tokens = lex(true, "/** @param string | int $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_11() {
  let tokens = lex(true, "/** @param (string | int) $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_12() {
  let tokens = lex(true, "/** @param string | int | float $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_13() {
  let tokens = lex(true, "/** @param string&int $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_14() {
  let tokens = lex(true, "/** @param string & int $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_15() {
  let tokens = lex(true, "/** @param (string & int) $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_16() {
  let tokens = lex(true, "/** @param (\n  string\n  &\n  int\n) $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_17() {
  let tokens = lex(true, "/** @param string & int & float $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_18() {
  let tokens = lex(true, "/** @param string & (int | float) $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_19() {
  let tokens = lex(true, "/** @param string | (int & float) $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_20() {
  let tokens = lex(true, "/** @param string & int | float $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_21() {
  let tokens = lex(true, "/** @param string | int & float $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_22() {
  let tokens = lex(true, "/** @param string[] $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_23() {
  let tokens = lex(true, "/** @param string [  ]  $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_24() {
  let tokens = lex(true, "/** @param (string | int | float)[] $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_25() {
  let tokens = lex(true, "/** @param string[][][] $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_26() {
  let tokens = lex(true, "/** @param string [  ] [][] $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_27() {
  let tokens = lex(true, "/** @param (((string | int | float)[])[])[] $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_28() {
  let tokens = lex(true, "/** @param $this $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_29() {
  let tokens = lex(true, "/** @param ?int $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_30() {
  let tokens = lex(true, "/** @param ?Foo<Bar> $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_31() {
  let tokens = lex(true, "/** @param array<int, Foo\\Bar> $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_32() {
  let tokens = lex(true, "/** @param array {'a': int} $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_33() {
  let tokens = lex(true, "/** @param array{a: int} $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_34() {
  let tokens = lex(true, "/** @param array{a: ?int} $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_35() {
  let tokens = lex(true, "/** @param array{a?: ?int} $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_36() {
  let tokens = lex(true, "/** @param array{0: int} $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_37() {
  let tokens = lex(true, "/** @param array{0?: int} $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_38() {
  let tokens = lex(true, "/** @param array{int, int} $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_39() {
  let tokens = lex(true, "/** @param array{a: int, b: string} $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_40() {
  let tokens = lex(
    true,
    "/** @param array{a?: int, b: string, 0: int, 1?: DateTime, hello: string} $foo */"
  ).unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_41() {
  let tokens = lex(true, "/** @param array{a: int, b: array{c: callable(): int}} $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_type_42() {
  let tokens = lex(true, "/** @param ?array{a: int} $foo */").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_no_var() {
  let tokens = lex(true, "/** @param string testing */\n$a = 5;").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn doc_param_description() {
  let tokens = lex(
    true,
    "/** @param string $foo testing
  *                    second line description*/\n$a = 5;"
  ).unwrap();
  insta::assert_yaml_snapshot!(tokens);
}
