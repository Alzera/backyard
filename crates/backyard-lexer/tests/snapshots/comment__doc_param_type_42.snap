---
source: crates/backyard-lexer/tests/comment.rs
expression: tokens
snapshot_kind: text
---
- token_type: CommentDocOpen
  value:
    - 47
    - 42
    - 42
  line: 1
  column: 0
  offset: 0
- token_type: DocTagParam
  value:
    - 64
    - 112
    - 97
    - 114
    - 97
    - 109
  line: 1
  column: 4
  offset: 4
- token_type: QuestionMark
  value:
    - 63
  line: 1
  column: 11
  offset: 11
- token_type: Type
  value:
    - 97
    - 114
    - 114
    - 97
    - 121
  line: 1
  column: 12
  offset: 12
- token_type: LeftCurlyBracket
  value:
    - 123
  line: 1
  column: 17
  offset: 17
- token_type: UnqualifiedName
  value:
    - 97
  line: 1
  column: 18
  offset: 18
- token_type: Colon
  value:
    - 58
  line: 1
  column: 19
  offset: 19
- token_type: Type
  value:
    - 105
    - 110
    - 116
  line: 1
  column: 21
  offset: 21
- token_type: RightCurlyBracket
  value:
    - 125
  line: 1
  column: 24
  offset: 24
- token_type: Variable
  value:
    - 36
    - 102
    - 111
    - 111
  line: 1
  column: 26
  offset: 26
- token_type: CommentDocClose
  value:
    - 42
    - 47
  line: 1
  column: 31
  offset: 31
